x/bank/simulation/decoder.go

```diff
@@ -7,14 +7,13 @@ import (
 	tmkv "github.com/tendermint/tendermint/libs/kv"
 
 	"github.com/cosmos/cosmos-sdk/codec"
-	"github.com/cosmos/cosmos-sdk/x/supply/keeper"
-	"github.com/cosmos/cosmos-sdk/x/supply/types"
+	"github.com/cosmos/cosmos-sdk/x/bank/types"
 )
 
 // DecodeStore unmarshals the KVPair's Value to the corresponding supply type
 func DecodeStore(cdc *codec.Codec, kvA, kvB tmkv.Pair) string {
 	switch {
-	case bytes.Equal(kvA.Key[:1], keeper.SupplyKey):
+	case bytes.Equal(kvA.Key[:1], types.SupplyKey):
 		var supplyA, supplyB types.Supply
 		cdc.MustUnmarshalBinaryBare(kvA.Value, &supplyA)
 		cdc.MustUnmarshalBinaryBare(kvB.Value, &supplyB)
```

Comment:
nit, line 24 should use `types.ModuleName`. We also need to add another case for coins stored

Reply:
We can't compare balances. They're stored as `{prefix}/{denom}/{coins}`.

Reply:
can you use `bytes.Prefix(kvA.Key, BalancesPrefix)`?

Reply:
Yes that's the key. I'm referring to the value --  what do you compare?

Reply:
You compare values only on the DiffKVStore function. DecodeStore just returns the unmarshaled values as strings

Reply:
> DecodeStore just returns the unmarshaled values as strings

This is what I'm stating -- you cannot unmarshal the values.

Reply:
> This is what I'm stating -- you cannot unmarshal the values.

and can you use the max denom length?

Reply:
no, the value is variable. You don't know how many bytes correspond to the dneom.